developer-docs/README.md

```diff
@@ -64,102 +64,162 @@ SEMATIC_API_ADDRESS: https://<my_prod1_host:port>
-`helm-charts` repo, and a modified `index.yaml`.  Commit and push both of
-these to a new branch, and create a PR for the change based on `gh-pages`
-necessary.
-
-Finally, draft the release on GitHub:
-- Add a "What's Changed" section.
-- Add a warning that the client version will need to be upgraded, if this applies.
-- Add a "New Contributors" section, if this applies.
-- Add a "Full Changelog" link.
-- Attach the wheel in the "Assets" section.
+1. Make the release PR, containing the previous changes. After implementing
+  comments and getting approval on the release PR, merge it and pull from the
+  updated `main`. It is mandatory to include this version of main in the
+  subsequent steps. If validation issues are discovered, they must be fixed
+  in patch PRs, and the process restarted from this step.
+
+1. Build the UI:
+    ```bash
+    $ make ui
+    ```
+
+1. Build the wheel:
+    ```bash
+    $ make wheel
+    ```
+
+1. Copy the wheel from `bazel-bin/sematic/sematic-*.whl` into a scratch
+  directory, and use a virtual env to test:
+
+    ```bash
+    $ pip3 install <wheel path>
+    $ sematic stop
+    $ sematic start
+    $ sematic run examples/mnist/pytorch
+    $ sematic stop
+    ```
+
+    Do this for all supported versions of Python. You can check your virtual env
+    Python version using `sematic version` (as well as the Server and Client
+    version).
+
+1. At this point, you should also deploy the Docker image to a cloud Dev environment
+  using the [internal Helm charts](/helm/sematic-server). To build the image, use
+  the same process you do when usually deploying dev code to a dev env. You can use
+  [`serve-dev`](https://github.com/sematic-ai/infrastructure/tree/main/bin)
+  to build and deploy the release candidate. 
+
+    1. Smoke test new features that were included or significantly updated in the
+    release.
+
+    1. Run the Testing Pipeline with "the works" on this deployment, and check that
+      it completes successfully, while perusing its outputs and logs to check they
+      render correctly.
+        ```bash
+        $ bazel run sematic/examples/testing_pipeline:__main__ -- \
+              --cloud \
+              --detach \
+              --max-parallelism 10 \
+              --inline \
+              --nested \
+              --no-input \
+              --sleep 10 \
+              --spam-logs 1000 \
+              --fan-out 10 \
+              --raise-retry 0.7 \
+              --external-resource \
+              --expand-shared-memory \
+              --cache-namespace test \
+              --images \
+              --virtual-funcs \
+              --fork-subprocess return 0 \
+              --fork-subprocess exit 0 \
+              --fork-subprocess exit 1 \
+              --fork-subprocess signal 2 \
+              --fork-subprocess signal 15
+        ```
+
+1. Test publishing the wheel. Check if the generated webpage is rendered
```

Comment:
It might be a little ambiguous on what `the generated webpages` are. Initially, I thought they are the dashboard  UI after installing and running the pip package. But they refer to the PyPI website, right?  Shall we add something to disambiguate?

Reply:
agree, not sure what to be looking for here.